app/src/main/java/com/konkuk/arabyte_aos/presentation/ui/component/ArabyteTopAppBar.kt

```diff
@@ -0,0 +1,80 @@
+package com.konkuk.arabyte_aos.presentation.ui.component
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.vectorResource
+import androidx.compose.ui.unit.dp
+import com.konkuk.arabyte_aos.R
+import com.konkuk.arabyte_aos.presentation.util.modifier.noRippleClickable
+import com.konkuk.arabyte_aos.ui.theme.ArabyteTheme
+
+/**
+ * 4가지 타입의 상단 앱 바를 나타내는 컴포넌트입니다.
+ *
+ * @param useBack Back 버튼 여부
+ * @param title App Bar의 가운데 Title
+ * @param optionalText App Bar의 우측 text (ex. 완료)
+ * @param optionalIconRes App Bar의 우측 버튼 (ex. 더보기 버튼)
+ */
+
+@Composable
+fun ArabyteTopAppBar(
+    modifier: Modifier = Modifier,
+    useBack: Boolean = true,
+    title: String? = null,
+    optionalText: String? = null,
+    optionalIconRes: Int? = null,
+    onBackClick: () -> Unit = {},
+    onOptionalClick: () -> Unit = {},
+) {
```

Comment:
```suggestion
@Composable
fun ArabyteTopAppBar(
    modifier: Modifier = Modifier,
    useBack: Boolean = true,
    title: String? = null,
    onBackClick: () -> Unit = {},
    onOptionalClick: () -> Unit = {},
)

fun ArabyteTopAppBar(
    modifier: Modifier = Modifier,
    useBack: Boolean = true,
    title: String? = null,
    optionalIconRes: Int,
    onBackClick: () -> Unit = {},
    onOptionalClick: () -> Unit = {},
)


@Composable
fun ArabyteTopAppBar(
    modifier: Modifier = Modifier,
    useBack: Boolean = true,
    title: String? = null,
    optionalText: String,
    onBackClick: () -> Unit = {},
    onOptionalClick: () -> Unit = {},
)

@Composable
private fun ArabyteTopAppBar(
    modifier: Modifier = Modifier,
    useBack: Boolean = true,
    title: String? = null,
    onBackClick: () -> Unit = {},
    optionComponent: @Composable () -> Unit = {},
)
```

전 이런식으로 overloading 하는 방식도 좋아보여요!
그러면 option 방식에서 icon과 text를 둘 다 사용하는데 icon이 무시되는 경우를 막을 수 있거든요!
즉 icon이나 text 방식 중 하나를 선택하게끔 강제하는 효과가 있습니다.